documentation/Concept guide.md

```diff
@@ -0,0 +1,157 @@
+- The [Component programming guide](https://spotify.github.io/HubFramework/component-programming-guide.html).
+- The [Content programming guide](https://spotify.github.io/HubFramework/content-programming-guide.html).
+
+## Content loading chain
+
+An array of [content operations](#content-operation) that are chained together to load the content for a view. A chain is automatically formed when a `HUBViewController` (or `HUBViewModelLoader`) is created with multiple content operations. In a chain, each operation is called sequentially, and its output state is transferred (by copying) to the next operation.
+
+For more information see:
+
+- The `HUBContentOperation` protocol.
+- The [Content programming guide](https://spotify.github.io/HubFramework/content-programming-guide.html).
+
+## Content operation
+
+An operation that is resposible for loading the content for a view. A single content operation can be resonsible for the entire content, or multiple ones can be chained together (to form a [content loading chain](#content-loading-chain)), each performing a specific mutation of the view's content.
+
+Content operations use a view model [builder](#builder) (`HUBViewModelBuilder`) to add, remove or change component models for a view. How they execute, and whether they load remote JSON or use local content is up to you. They are defined using the `HUBContentOperation` protocol (and optionally, sub-protocols).
+
+For more information see:
+
+- The `HUBContentOperation` protocol.
+- The `HUBViewModelBuilder` protocol.
+- The [Content programming guide](https://spotify.github.io/HubFramework/content-programming-guide.html).
+
+## Factory
+
+An object that is responsible for creating implementations of a certain protocol. The [factory pattern](https://en.wikipedia.org/wiki/Factory_method_pattern) is used extensively throughout the Hub Framework to avoid shared state. By always creating unique object instances for each context - no state is ever shared. While this requires a bit more code to be written (you have to write both an implementation **and a factory**, for example for [components](#component)), it makes the framework a lot more predictable and less error prone.
+
+For more information see:
+
+- ["Factory method pattern" on Wikipedia](https://en.wikipedia.org/wiki/Factory_method_pattern).
+- The  `HUBComponentFactory` protocol.
+- The `HUBActionFactory` protocol.
+- The `HUBViewControllerFactory` protocol.
+- The `HUBViewModelLoaderFactory` protocol.
+
+## Layout trait
+
+Layout traits are used to define layout relationships between components. Instead of hard-coding margins and relationships between concrete component implementations - layout traits enables components to work with layout in a more abstract way. Each component simply defines which layout traits that best describe it, in terms of layout, and then an implementation of `HUBComponentLayoutManager` computes the exact margins that should be applied to it.
```

Comment:
Either `Each component simply defines which layout traits best describe it` or `Each component simply defines the layout traits that best describe it`